src/wp-includes/template.php

```diff
@@ -965,50 +965,138 @@ function wp_finalize_template_enhancement_output_buffer( string $output, int $ph
 
 	$filtered_output = $output;
 
-	/**
-	 * Filters the template enhancement output buffer prior to sending to the client.
-	 *
-	 * This filter only applies the HTML output of an included template. This filter is a progressive enhancement
-	 * intended for applications such as optimizing markup to improve frontend page load performance. Sites must not
-	 * depend on this filter applying since they may opt to stream the responses instead. Callbacks for this filter are
-	 * highly discouraged from using regular expressions to do any kind of replacement on the output. Use the HTML API
-	 * (either `WP_HTML_Tag_Processor` or `WP_HTML_Processor`), or else use {@see DOM\HtmlDocument} as of PHP 8.4 which
-	 * fully supports HTML5.
-	 *
-	 * Important: Because this filter is applied inside an output buffer callback (i.e. display handler), any callbacks
-	 * added to the filter must not attempt to start their own output buffers. Otherwise, PHP will raise a fatal error:
-	 * "Cannot use output buffering in output buffering display handlers."
-	 *
-	 * @since 6.9.0
-	 *
-	 * @param string $filtered_output HTML template enhancement output buffer.
-	 * @param string $output          Original HTML template output buffer.
-	 */
-	$filtered_output = (string) apply_filters( 'wp_template_enhancement_output_buffer', $filtered_output, $output );
+	$did_just_catch_exception = false;
+
+	$error_log = array();
+	set_error_handler(
+		static function ( int $level, string $message, ?string $file = null, ?int $line = null ) use ( &$error_log, &$did_just_catch_exception ) {
+			// Switch a user error to an exception so that it can be caught and the buffer can be returned.
+			if ( E_USER_ERROR === $level ) {
+				throw new Exception( __( 'User error triggered:' ) . ' ' . $message );
+			}
+
+			// Display a caught exception as an error since it prevents any of the output buffer filters from applying.
+			if ( $did_just_catch_exception ) { // @phpstan-ignore if.alwaysFalse (The variable is set in the catch block below.)
+				$level = E_USER_ERROR;
+			}
+
+			// Capture a reported error to be displayed by appending to the processed output buffer if display_errors is enabled.
+			if ( error_reporting() & $level ) {
+				$error_log[] = compact( 'level', 'message', 'file', 'line' );
+			}
+			return false;
+		}
+	);
+	$display_errors = ini_get( 'display_errors' );
+	if ( $display_errors ) {
```

Comment:
if this is `stderr` is it a problem for the output buffering?

Reply:
Oh, interesting. I wasn't aware that was an [option](https://www.php.net/manual/en/errorfunc.configuration.php#ini.display-errors). I suppose it should be then be:

```suggestion
	if ( $display_errors && 'stderr' !== $display_errors ) {
```

Reply:
Actually, I just tried this and it still is going to the frontend:

```php
add_action( 'init', function () {
	ini_set( 'display_errors', 'stderr' );
} );

add_action( 'wp_footer', function () {
	trigger_error( 'You are not doing this right!!', E_USER_WARNING );
} );
```

Result:

```html
...
		</svg>

		<br />
<b>Warning</b>:  You are not doing this right!! in <b>/var/www/src/wp-content/plugins/try-display-errors-stderr.php</b> on line <b>11</b><br />
<script type="importmap" id="wp-importmap">
...
```

This is in the wordpress-develop environment. 

However, when I use PHP's built-in web server, when run this PHP coe:

```php
ini_set( 'display_errors', 'stderr' );
trigger_error( 'You are not doing this right!!', E_USER_WARNING );
```

The result is that the error is _not_ emitted to the frontend. It is sent to the terminal after the regular log entry:

```
[Sat Nov  1 19:59:25 2025] PHP Warning:  You are not doing this right!! in /private/tmp/try-stderr/try.php on line 4
[Sat Nov  1 19:59:25 2025] PHP Stack trace:
[Sat Nov  1 19:59:25 2025] PHP   1. {main}() /private/tmp/try-stderr/try.php:0
[Sat Nov  1 19:59:25 2025] PHP   2. trigger_error($message = 'You are not doing this right!!', $error_level = 512) /private/tmp/try-stderr/try.php:4
<br />
<font size='1'><table class='xdebug-error xe-warning' dir='ltr' border='1' cellspacing='0' cellpadding='1'>
<tr><th align='left' bgcolor='#f57900' colspan="5"><span style='background-color: #cc0000; color: #fce94f; font-size: x-large;'>( ! )</span> Warning: You are not doing this right!! in /private/tmp/try-stderr/try.php on line <i>4</i></th></tr>
<tr><th align='left' bgcolor='#e9b96e' colspan='5'>Call Stack</th></tr>
<tr><th align='center' bgcolor='#eeeeec'>#</th><th align='left' bgcolor='#eeeeec'>Time</th><th align='left' bgcolor='#eeeeec'>Memory</th><th align='left' bgcolor='#eeeeec'>Function</th><th align='left' bgcolor='#eeeeec'>Location</th></tr>
<tr><td bgcolor='#eeeeec' align='center'>1</td><td bgcolor='#eeeeec' align='center'>0.0001</td><td bgcolor='#eeeeec' align='right'>359408</td><td bgcolor='#eeeeec'>{main}(  )</td><td title='/private/tmp/try-stderr/try.php' bgcolor='#eeeeec'>.../try.php<b>:</b>0</td></tr>
<tr><td bgcolor='#eeeeec' align='center'>2</td><td bgcolor='#eeeeec' align='center'>0.0001</td><td bgcolor='#eeeeec' align='right'>359816</td><td bgcolor='#eeeeec'><a href='http://www.php.net/function.trigger-error' target='_new'>trigger_error</a>( <span>$message = </span><span>&#39;You are not doing this right!!&#39;</span>, <span>$error_level = </span><span>512</span> )</td><td title='/private/tmp/try-stderr/try.php' bgcolor='#eeeeec'>.../try.php<b>:</b>4</td></tr>
</table></font>
```

So there is an inconsistency here. 

It seems better safe to not emit in the case of `stderr` than to assume it is OK. So I've updated the logic in 5b13a71d53 to _not_ append the errors when it is `stderr`.

Reply:
there _might_ be some difference when running as `php -ddisplay_errors=stderr` than when calling `ini_set( 'display_errors', 'stderr' )` based on a note in the PHP manual page, but I think that’s only saying we don’t trap fatals because we’re catching them in the environment in which we’re throwing them, so probably doesn’t impact where the output is set. it wold not be the first time I’ve found the PHP manual page to be incomplete or incorrect in its assertions though.

~I wish I understood the potential values here better.~ Seems like there are only two modes: `STDOUT` and `STDERR`, and you made the proper detection, because the INI value is either exactly `stderr` (in which case it uses the `STDERR` mode) or it’s _anything else_ in which case it’s `STDOUT`.

https://github.com/php/php-src/blob/a979e9f897a90a580e883b1f39ce5673686ffc67/main/main.c#L489-L518